modules/signals/src/state-source.ts

```diff
@@ -10,16 +10,26 @@ import {
 } from '@angular/core';
 import { Prettify } from './ts-helpers';
 
-const STATE_WATCHERS = new WeakMap<Signal<object>, Array<StateWatcher<any>>>();
+declare const ngDevMode: unknown;
+
+const STATE_WATCHERS = new WeakMap<object, Array<StateWatcher<any>>>();
 
 export const STATE_SOURCE = Symbol('STATE_SOURCE');
 
 export type WritableStateSource<State extends object> = {
-  [STATE_SOURCE]: WritableSignal<State>;
+  [STATE_SOURCE]: {
+    [Property in keyof State]: WritableSignal<State[Property]>;
+  };
 };
 
 export type StateSource<State extends object> = {
-  [STATE_SOURCE]: Signal<State>;
+  [STATE_SOURCE]: { [Property in keyof State]: Signal<State[Property]> };
+};
+
+export type StateResult<StateInput extends object> = {
```

Comment:
I wonder if this this should also be exported.

Reply:
I’m not sure. What criteria do you usually follow for exposing types?
If I don’t see an immediate need, I tend to hold off and wait for a concrete use case.